.github/workflows/update-terraform-submodule.yaml

```diff
@@ -0,0 +1,42 @@
+name: Update Aiven Terraform Provider submodule to sync code samples
+on:
+  schedule:
+    - cron: '0 8 1,15 * *' # Runs at 08:00 on the 1st and 15th every month
```

Comment:
nit: maybe run this more often? There is also the `workflow_dispatch` to run this manually which is good